### PR TITLE
Support libbacktrace on Linux

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
None

#### Purpose of change
We've supported libbacktrace as an optional dependency for better backtraces on Windows for a while now.  At the same time, our backtrace implementation on Linux is very inefficient.

#### Describe the solution
Optionally use libbacktrace on Linux too, when requested at configure time.  This is much less code and much faster at runtime, but we retain the old implementation as well for the time being so that we are not adding an additional mandatory dependency.

Currently the LIBBACKTRACE option is off by default; it would be nice to make it on at least for the release builds.

#### Describe alternatives you've considered
Trying to test on MacOS too?

#### Testing
Deliberately causing an error in-game yields a backtrace similar to this:

```
13:17:05.078 ERROR : (error message will follow backtrace)
    0x86f03d    …/src/debug.cpp:941    bt_full(backtrace_state*, int, std::function<int (unsigned long, char const*, int, char const*)> const&, std::function<void (char const*, int)> const&)
    0x86eecf    …/src/debug.cpp:1200    debug_write_backtrace(std::ostream&)
    0x86e6a2    …/src/debug.cpp:1440    DebugLog(DebugLevel, DebugClass)
    0x86e154    …/src/debug.cpp:505    realDebugmsg(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    0x216a94    …/src/debug.h:78    void realDebugmsg<>(char const*, char const*, char const*, char const*)
    0x16d0bb7    …/src/panels.cpp:410    panel_manager::get_current_layout()
    0x16d0eea    …/src/panels.cpp:453    panel_manager::init()
    0xa9c198    …/src/game.cpp:689    game::setup()
    0xfd3e7d    …/src/main_menu.cpp:1025    main_menu::load_character_tab(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    0xfd2bc5    …/src/main_menu.cpp:841    main_menu::opening_screen()
    0x1c70f2b    …/src/main.cpp:763    main
    0x7f6dd6da1319    [unknown src]:0    [unknown func]
    0x7f6dd6da13cb    [unknown src]:0    [unknown func]
    0x211460    [unknown src]:0    [unknown func]
    0xffffffffffffffff    [unknown src]:0    [unknown func]
Backtrace emission took 0 seconds.
```

#### Additional context
I'm also the one responsible for our existing (very slow) backtrace implementation on Linux, and it annoys me often while debugging the game that it's so slow.  So, I feel an obligation to try to fix this issue.

This has a good chance of breaking something in the build matrix, so I'm opening as a draft while I check out how the CI fares.

### CODE REVIEW

File: build-scripts/requirements.sh [hunk 1]
```diff
@@ -38,6 +38,17 @@ fi
 set -x
 if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
     $travis_retry sudo apt-get --yes install parallel
+
+    if [[ "$LIBBACKTRACE" == "1" ]]; then
+        git clone https://github.com/ianlancetaylor/libbacktrace.git
+        (
+            cd libbacktrace
+            git checkout 4d2dd0b172f2c9192f83ba93425f868f2a13c553
```
Comment (casswedson) [inline build-scripts/requirements.sh:46]: Why do a `checkout` here if this commit is the latest on ianlancetaylor's repo
https://github.com/ianlancetaylor/libbacktrace/commit/4d2dd0b172f2c9192f83ba93425f868f2a13c553

**Comment by andrei8l** [comment]
I'm getting unusable logs with libbacktrace and release builds
<details>
<summary>with libbacktrace</summary>

built with `make BACKTRACE=1 LIBBACKTRACE=1 RELEASE=1`
```gdb
The program has crashed.
See the log file for a stack trace.
CRASH LOG FILE: ./config/crash.log
VERSION: 0.F-9700-g36a08cc19e-dirty
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:

    0x55e.d07.dda.2c4    [unknown src]:0    [unknown func]
    0x55e.d07.dbe.2c2    [unknown src]:0    [unknown func]
    0x55e.d07.dbe.50b    [unknown src]:0    [unknown func]
    0x7f7.0ac.23e.8df    [unknown src]:0    [unknown func]
    0x7f7.0ac.28e.36c    [unknown src]:0    [unknown func]
    0x7f7.0ac.23e.837    [unknown src]:0    [unknown func]
    0x55e.d07.dfe.d20    [unknown src]:0    [unknown func]
    0x55e.d07.f96.b9c    [unknown src]:0    [unknown func]
    0x55e.d07.f98.602    [unknown src]:0    [unknown func]
    0x55e.d07.e3a.e6e    [unknown src]:0    [unknown func]
    0x55e.d07.921.330    [unknown src]:0    [unknown func]
    0x7f7.0ac.229.28f    [unknown src]:0    [unknown func]
    0x7f7.0ac.229.349    [unknown src]:0    [unknown func]
    0x55e.d07.adb.9a4    ../sysdeps/x86_64/start.S:115    [unknown func]
    0xf.fff.fff.fff.fff.fff    [unknown src]:0    [unknown func]
```
</details>
<details>
<summary>without libbacktrace</summary>

built with `make BACKTRACE=1 LIBBACKTRACE=0 RELEASE=1`
```gdb
The program has crashed.
See the log file for a stack trace.
CRASH LOG FILE: ./config/crash.log
VERSION: 0.F-9700-g36a08cc19e-dirty
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:

    ./cataclysm(debug_write_backtrace(std::ostream&)+0x3d) [0x55ba799a3e22]
    ./cataclysm(+0xba3303) [0x55ba79985303]
    ./cataclysm(+0xba354c) [0x55ba7998554c]
    /usr/lib/libc.so.6(+0x3e8e0) [0x7f551763e8e0]
    /usr/lib/libc.so.6(+0x8e36c) [0x7f551768e36c]
    /usr/lib/libc.so.6(raise+0x18) [0x7f551763e838]
    ./cataclysm(debug_menu::debug()+0x2795) [0x55ba799d128b]
    ./cataclysm(game::do_regular_action(action_id&, avatar&, cata::optional<tripoint> const&)+0x3c67) [0x55ba79b690eb]
    ./cataclysm(game::handle_action()+0xa7f) [0x55ba79b6ab51]
    ./cataclysm(do_turn()+0x5a4) [0x55ba79a0d3d9]
    ./cataclysm(main+0x1832) [0x55ba794e8363]
    /usr/lib/libc.so.6(+0x29290) [0x7f5517629290]
    /usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7f551762934a]
    ./cataclysm(_start+0x25) [0x55ba796a29e5]

    Attempting to repeat stack trace using debug symbols…
    debug_write_backtrace(std::ostream&)
    ??:?
    log_crash
    crash.cpp:?
    signal_handler
    crash.cpp:?
    ??
    ??:0
    ??
    ??:0
    ??
    ??:0
    debug_menu::debug()
    ??:?
    game::do_regular_action(action_id&, avatar&, cata::optional<tripoint> const&)
    ??:?
    game::handle_action()
    ??:?
    do_turn()
    ??:?
    main
    ??:?
    ??
    ??:0
    ??
    ??:0
    _start
    …/src/glibc/csu/../sysdeps/x86_64/start.S:117
```
</details>

with libbacktrace built from [AUR](https://aur.archlinux.org/packages/libbacktrace-git)

**Comment by jbytheway** [comment]
Looks like the CI changes are working as expected, so I think this is now good to go.  This is certainly helping me in my dev work.  Whether we want to change this for Linux release binaries as well is something that can be decided in the future.

**Comment by jbytheway** [comment]
So it turns out this PR did in fact have the side-effect of turning on `-g1` for the mingw cross-compile which we use as the release build, so that is now larger (total zip size is 131MB, up from 91MB; executable size 314MB up from 79MB).  I think this is a good thing, because backtraces in bug reports from those folks should hopefully be more useful now.  But I wanted to mention it in case this is considered a regression.

**Pull request merged**